qt/applications/workbench/workbench/plotting/figureinteraction.py

```diff
@@ -686,6 +686,15 @@ def _toggle_normalization(self, ax):
 
             ax.update_waterfall(0, 0)
 
+        # The colorbar can get screwed up with ragged workspaces and log scales as they go
+        # through the normalisation toggle.
+        # Set it to Linear and change it back after if necessary, since there's no reason
+        # to duplicate the handling.
+        colorbar_log = False
+        if ax.images:
+            colorbar_log = isinstance(ax.images[-1].norm, LogNorm)
+        self._change_colorbar_axes(Normalize)
```

Comment:
This line could be put inside the above if-statement since it isn't needed for non-image plots. You could also add an `if colorbar_log` check since you don't need to change anything if the colorbar is already linearly scaled.